lib/ReactViewModels/ViewState.ts

```diff
@@ -592,3 +582,10 @@ export default class ViewState {
     return this.currentTool !== undefined;
   }
 }
+
+interface Tool {
+  toolName: string;
+  getToolComponent: () => React.Component | Promise<React.Component>;
+  showCloseButton: boolean;
+  params: any;
```

Comment:
```suggestion
  params?: any;
```
Should params be an optional parameter